pkg/opensearch/client/search_request_test.go

```diff
@@ -179,28 +242,43 @@ func Test_Given_new_search_request_builder_for_es_OpenSearch_1_0_0(t *testing.T)
 		aggBuilder.Terms("1", "@hostname", nil)
 		aggBuilder.DateHistogram("2", "@timestamp", nil)
 
+		// adding nested terms aggregations
+		aggBuilder.Terms("aggregation_name", "field_name", func(a *TermsAggregation, b AggBuilder) {
+			b.Terms("inner_aggregation_name", "field_name.inner_field", nil)
+		})
+
 		sr, err := b.Build()
 		assert.NoError(t, err)
 
 		aggs := sr.Aggs
-		assert.Len(t, aggs, 2)
+		assert.Len(t, aggs, 3)
 		assert.Equal(t, "1", aggs[0].Key)
 		assert.Equal(t, "terms", aggs[0].Aggregation.Type)
 		assert.Equal(t, "2", aggs[1].Key)
 		assert.Equal(t, "date_histogram", aggs[1].Aggregation.Type)
 
+		// test nested terms aggregation
+		assert.Equal(t, "aggregation_name", aggs[2].Key)
+		assert.Equal(t, "terms", aggs[2].Aggregation.Type)
+		assert.Equal(t, "field_name", aggs[2].Aggregation.Aggregation.(*TermsAggregation).Field)
+		assert.Equal(t, "inner_aggregation_name", aggs[2].Aggregation.Aggs[0].Key)
+		assert.Equal(t, "terms", aggs[2].Aggregation.Aggs[0].Aggregation.Type)
+		assert.Equal(t, "field_name.inner_field", aggs[2].Aggregation.Aggs[0].Aggregation.Aggregation.(*TermsAggregation).Field)
+
 		t.Run("When marshal to JSON should generate correct json", func(t *testing.T) {
 			body, err := json.Marshal(sr)
 			assert.NoError(t, err)
 			json, err := simplejson.NewJson(body)
 			assert.NoError(t, err)
 
-			assert.Len(t, json.Get("aggs").MustMap(), 2)
+			assert.Len(t, json.Get("aggs").MustMap(), 3)
 			assert.Equal(t, "@hostname", json.GetPath("aggs", "1", "terms", "field").MustString())
 			assert.Equal(t, "@timestamp", json.GetPath("aggs", "2", "date_histogram", "field").MustString())
+
+			// test for nested terms aggregations
+
 		})
 	})
-
```

Comment:
This line shouldn't be deleted. (as a style thing, it nice to have a new line between `t.Run` calls)